plugins/woocommerce/src/Internal/Admin/Settings/PaymentsController.php

```diff
@@ -306,11 +308,26 @@ function ( $dismissal ) {
 	}
 
 	/**
-	 * Check if the WooPayments plugin is active.
+	 * Check if the WooPayments account is onboarded.
 	 *
 	 * @return boolean
 	 */
-	private function is_woopayments_active(): bool {
-		return class_exists( '\WC_Payments' );
+	private function is_woopayments_account_onboarded(): bool {
+		// If WooPayments is active right now, we will not get to this point since the plugin is active check is done first.
+		if ( ! class_exists( '\WC_Payments' ) ) {
+			return false;
+		}
+
+		$account_data = get_option( 'wcpay_account_data', array() );
+		if ( empty( $account_data['data']['account_id'] ) ) {
+			return false;
+		}
+
+		if ( empty( $account_data['data']['details_submitted'] ) ) {
```

Comment:
TIL this also checks if the value is false!

Reply:
It is a catchall way to check for not-defined (deeply) and any [falsy value](https://www.php.net/manual/en/language.types.boolean.php#language.types.boolean.casting).